pkg/trace/api/api.go

```diff
@@ -498,8 +507,9 @@ func getTraces(v Version, w http.ResponseWriter, req *http.Request) (pb.Traces,
 		// in v01 we actually get spans that we have to transform in traces
 		var spans []pb.Span
 		if err := json.NewDecoder(req.Body).Decode(&spans); err != nil {
-			log.Errorf("Cannot decode %s traces payload: %v", v, err)
 			httpDecodingError(err, []string{tagTraceHandler, fmt.Sprintf("v:%s", v)}, w)
+			atomic.AddInt64(&ts.TracesDropped.DecodingError, traceCount)
```

Comment:
Note that we don't actually need to report `metrics.Count("datadog.trace_agent.receiver.decoding_error", 1, nil, 1)` here as it turns out `httpDecodingError` is already reporting a metric `datadog.trace_agent.receiver.error` tagged by `error:decoding-error`. So in the case of `traceCount=0` for a missing trace count header, we still have that metric to track the number of decoding errors.

Reply:
True 👍 We should remove it and possibly update our RFC.

Reply:
Ok, renamed "traces malformed" to "spans malformed" and updated RFC.